### DISCUSSION TITLE
Welcome to RDDRONE-BMS772 Discussions!

### DISCUSSION THREAD

**Discussion by igalloway** [NXP-Robotics/RDDRONE-BMS772#2] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by dd-rrc-ps** [comment]
Hi, 
after setting the configuration of the BMS for normal use, i got error in the step "make the binary with make". 
The Log: 
```
make[3]: Entering directory '/home/ddao/projects/apps/canutils/libcanardv1'
Unpacking: libcanard-.zip -> libcanard-
Archive:  libcanard-.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of libcanard-.zip or
        libcanard-.zip.zip, and cannot find libcanard-.zip.ZIP, period.
make[3]: *** [Makefile:61: libcanard-] Error 9
make[3]: Leaving directory '/home/ddao/projects/apps/canutils/libcanardv1'
make[2]: *** [Makefile:61: /home/ddao/projects/apps/canutils/libcanardv1_context] Error 2
make[2]: Leaving directory '/home/ddao/projects/apps'
make[1]: *** [Makefile:163: context] Error 2
make[1]: Leaving directory '/home/ddao/projects/apps'
make: *** [tools/Makefile.unix:337: context] Error 2
```
Can you please check the libcanard-.zip file?

**Reply by cisvanmierlo** [reply]
Hi dd-rrc-ps,

It looks like it is trying to unzip a non-existing archive.
Maybe something went wrong during the installation.

Could you check if the patch file 0001-apps-patch-BMS4.0.patch was applied correctly?

In this part of the patch file the correct hash will be added for libcanard: 
https://github.com/NXPHoverGames/RDDRONE-BMS772/blob/main/Patchfiles/0001-apps-patch-BMS4.0.patch#L227-L233

Which will result in unzipping from the following link (this link will download a zip file): 
https://github.com/UAVCAN/libcanard/archive/cde670347425023480a1417fcd603b27c8eb06c1.zip

**Reply by dd-rrc-ps** [reply]
Thank you for your reply. I removed everything and start again from the start, and it worked. I think the last download was corrupted somehow.

**Comment by OgzAvn** [comment]
Hello, I'm just getting to know the RD DRONE - BMS 772 board.
I connected a 6S battery pack to the board. I can see all the values with ToolBoxs provided by NXP via matlab/simulink. I am reading the values from FREEMASTER.
I want to run the SSD1306 oled display on the card. 3.3 volts from SBC to the screen, but it does not work. Why is 5V not coming from SBC???? The display communicates with I2C. Can you help me with this ?

**Reply by cisvanmierlo** [reply]
Hello @OgzAvn,

This example in NuttX does not support FREEMASTER.
To get the battery values, you could use UART (the CLI), (UAV)CAN, I2C, NFC and the display.
How to configure these communications can be found in the release notes.

Display support is added in the latest 5.0 release, make sure the board is updated to this latest release.
Keep in mind that currently only 3.3V for the display is supported.
Make sure D34 is placed and D35 is not placed. This will ensure 3.3V supply on J23 together with I2C lines for driving an SSD1306 type display.
This because the 5V needs to be enabled during (NuttX) startup to initialize the display.
You could configure the SBC differently to enable the 5V in all SBC modes.

**Reply by OgzAvn** [reply]
Hi @cisvanmierlo 

Thank you for your reply !!

I swapped the D34 with the D35. The screen is now 3.3Volt, but it does not work. I was able to see the values in FreeMaster before. Can I use the screen on this card only in NuutX? When I put the SBC in normal mode, 3.3V is coming, but not 5V. Do I need to change the software of SBC?

Best Regards!!

**Reply by cisvanmierlo** [reply]
Hi @OgzAvn, 

Could it be that using this freemaster tool, the microcontroller got re-programmed and this NuttX BMS example is replaced?
I would suggest to make sure the following is programmed:
https://github.com/NXPHoverGames/RDDRONE-BMS772\
Documentation, source code, binary and manual can all be found at that repository.

The latest state should include display support for 3.3V.
If needed, you could look at the source files in that git repository and see how the display is programmed using the RTOS NuttX.

**Reply by OgzAvn** [reply]
Hi @cisvanmierlo ,

Thank you !

**Reply by OgzAvn** [reply]
hello me again :)

I connected a 24Volt 150mA load to the output of the board. I'm measuring current, but the exact value isn't coming out. I'm reading on FreeMaster. I checked the registers. I wrote the shunt resistor at the absolutely correct value. How should I write a C function to get the correct value. Or do you have another recommendation?

Best Regards!!

**Reply by cisvanmierlo** [reply]
Hi @OgzAvn,

The provided code in this git repository does not support FreeMASTER and I do not have experience with this tool.
If your questions are specific to FreeMASTER , I would recommend you to check out the NXP community about FreeMASTER. 
There my colleagues, who are experts on FreeMASTER , could provide you the best support for this tool.
Check out https://community.nxp.com/t5/FreeMASTER/bd-p/freemaster for the FreeMASTER support portal.

I could point you to the C code that is used to read and convert the current for this board in the software example.
If you look at https://github.com/NXPHoverGames/RDDRONE-BMS772/blob/main/src/BCC/bcc_monitoring.c#L376-L401.
You can see that first a measurement is done, then the correct current registers are read, this is then converted to the current value as a floating point in mA and later on this is converted to A.
The BCC_GET_ISENSE_AMP() is a macro that is defined in https://github.com/NXPHoverGames/RDDRONE-BMS772/blob/main/inc/BCC/Derivatives/bcc.h#L163.
 
Hopefully this will help!
Kind regards,

Cis van Mierlo

**Reply by OgzAvn** [reply]
Hi, @cisvanmierlo 

Thank you I solved. 

Best Rigards!
